sql-plugin/src/main/scala/com/nvidia/spark/rapids/SortUtils.scala

```diff
@@ -210,6 +211,23 @@ class GpuSorter(
     }
   }
 
+  private[this] lazy val hasNestedInKeyColumns = cpuOrderingInternal.exists { order =>
+    projectedBatchTypes(order.child.asInstanceOf[BoundReference].ordinal) match {
+      case _: ArrayType | _: StructType | _: MapType => true
+      case _ => false
+    }
+  }
```

Comment:
This should leverage `DataTypeUtils.isNestedType`.  Also, do we need to go all the way through the cpuOrdering, downcasting to a bound reference, and indexing the projected batches?  It seems like we can simply examine the original sort order directly, but maybe I'm missing something.  e.g.:
```suggestion
  private[this] lazy val hasNestedInKeyColumns = sortOrder.exists { order =>
    DataTypeUtils.isNestedType(order.child.dataType)
  }
```

Reply:
Thanks for review. Updated to leverage the `DataTypeUtils.isNestedType`. And yes, you are right, we'd better to go into a bound reference.

As far as I know, since here we check nested existence in the keys for the `Table.merge`, ideally `cudfOrdering` should be used, instead of the original sort order.

However the `OerderByArg`  has no type info, so use the `cpuOrderingInternal` instead which has the data type, and contains the same key columns with `cudfOrdering`.